### ISSUE TITLE
strict() ignores options that contain -

### ISSUE THREAD

**Issue by Linard11** [yargs/yargs#1171]
`test.js`

```
const yargs = require('yargs');
yargs.strict();
yargs.showHelpOnFail(false);
console.log('test-flag: ' + yargs.argv['test-flag']);
```

**Expected behavior:**
```
node test.js --test-flag

Unknown argument: test-flag
```

**Actual behavior:**
```
node test.js --test-flag

test-flag: true
```

_Though it works fine when given a option without a `-`:_
```
node test.js --test

Unknown argument: test
```

**Comment by begin-again** [comment]
Still occurring in v12.0.5

**Comment by juergba** [comment]
see https://github.com/yargs/yargs/issues/1325#issuecomment-477487511

**Comment by bcoe** [comment]
This should now be addressed in `yargs@14.2.0` 🎉, please let us know if this continues to be an issue.
